web/packages/teleport/src/Discover/Kubernetes/EnrollEKSCluster/EnrollEksCluster.tsx

```diff
@@ -0,0 +1,421 @@
+        setTableData({ ...data, fetchStatus: 'disabled' });
+        return;
+      }
+
+      // Check if fetched EKS clusters have a Kube
+      // server for it, to prevent user from enrolling
+      // the same cluster.
+      const query = fetchedEKSClusters
+        .map(cluster => `name == "${cluster.name}"`)
+        .join(' || ');
+      const existingKubeServers = await fetchKubeServers(query, 0);
+      const checkedEksClusters: CheckedEksCluster[] = fetchedEKSClusters.map(
+        cluster => {
+          let serverExists = existingKubeServers.agents.some(k => {
+            const regionLabel = k.labels?.find(l => l.name === 'region')?.value;
+            const accountLabel = k.labels?.find(
+              l => l.name === 'account-id'
+            )?.value;
+
+            return (
+              cluster.name === k.name &&
+              cluster.region === regionLabel &&
+              cluster.accountId === accountLabel
+            );
+          });
+
+          return {
+            ...cluster,
+            kubeServerExists: serverExists,
+          };
+        }
+      );
+
+      setFetchClustersAttempt({ status: 'success' });
+      setTableData({
+        currRegion: data.currRegion,
+        startKey: nextToken,
+        fetchStatus: nextToken ? '' : 'disabled',
+        // concat each page fetch.
+        items: [...data.items, ...checkedEksClusters],
+      });
+    } catch (err) {
+      const errMsg = getErrMessage(err);
+      setFetchClustersAttempt({ status: 'failed', statusText: errMsg });
+      setTableData(data); // fallback to previous data
+      emitErrorEvent(`database fetch error: ${errMsg}`);
+    }
+  }
+
+  function clear() {
+    if (fetchClustersAttempt.status === 'failed') {
+      setFetchClustersAttempt({ status: '' });
+    }
+    if (tableData.items.length > 0) {
+      setTableData(emptyTableData);
+    }
+    if (selectedCluster) {
+      setSelectedCluster(null);
+    }
+    setEnrollmentState({ status: 'notStarted' });
+  }
+
+  function fetchKubeServers(query: string, limit: number) {
+    return ctx.kubeService.fetchKubernetes(ctx.storeUser.getClusterId(), {
+      query,
+      limit,
+    });
+  }
+
+  async function handleOnEnroll() {
+    const integrationName = (agentMeta as EksMeta).awsIntegration.name;
+    setEnrollmentState({ status: 'enrolling' });
+
+    try {
+      const response = await integrationService.enrollEksClusters(
+        integrationName,
+        {
+          region: selectedRegion,
+          enableAppDiscovery: isAppDiscoveryEnabled,
+          joinToken: joinToken.id,
+          resourceId: joinToken.internalResourceId,
+          clusterNames: [selectedCluster.name],
+        }
+      );
+
+      const result = response.results?.find(
+        c => c.clusterName === selectedCluster.name
+      );
+      if (!result) {
+        setEnrollmentState({
+          status: 'error',
+          error: `Cluster "${selectedCluster.name}" enrollment result is unknown.`,
```

Comment:
normally there is a `re-try` button on the dialog, so we don't have to say the following for each error (and less clicks for user) `Close the dialog, and try again`.

also, should all the error cases emit an error event?

Reply:
Added retry button - https://github.com/gravitational/teleport/pull/36634/commits/eeb51d371050b50fd9d6d98d5b483db4f08c855a

And also added emitting of the error events - https://github.com/gravitational/teleport/pull/36634/commits/e09b55fa9fbf6edecf5b19569107f503f03c2fbf